src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -77,8 +78,12 @@ public async Task<IEnumerable<Type>> GetExtensionsStartupTypesAsync()
                     {
                         bindingsSet.Add(binding.Type);
                     }
+                    isPrecompiledFunctionApp = isPrecompiledFunctionApp || functionMetadata.Language == DotNetScriptTypes.DotNetAssembly;
                 }
+            }
 
+            if (bundleConfigured && (!isPrecompiledFunctionApp || _extensionBundleManager.IsLegacyExtensionBundle()))
```

Comment:
I'd like to see an `else` with a log output here explaining that we're not loading the bundles (and why).

Reply:
+1 for adding more logging